### ISSUE TITLE
VsCode freeze when changing git branch

### ISSUE THREAD

**Issue by ggrimbert** [microsoft/vscode#63919]
Hi,
Sorry if this is a duplicate but I haven't find any bug like this one.

- VSCode Version: 1.29.1
- OS Version: Ubuntu 18.04 KDE

Steps to Reproduce:

1. Open vsCode on a project with git
2. Modify some code and save
3. Go to scm tab and try to change the branch currently in use

A modal window opens to say that my modifications will be erased (I'm using vscode in French): 
![image](https://user-images.githubusercontent.com/15869188/49154435-a81c6d00-f318-11e8-8dec-5c93db476277.png)

This modal and the layout block the totality of my system. My cursor is responding to move; but not to click and I can't change the window currently displayed. The only way to get my system back is to connect from another computer on mine and kill the vscode process.

Does this issue occur when all extensions are disabled?: Yes

**Comment by joaomoreno** [comment]
Could you record a screen recording?

Can you reproduce with `code --disable-extensions`?

How did you install VS Code?

**Comment by ggrimbert** [comment]
The video is in the archive attached. I had to pause one time which is why the video jumped.
[VsCodeVideo.zip](https://github.com/Microsoft/vscode/files/2642970/VsCodeVideo.zip)

Steps to reproduce : 
- Create new folder
- Open vscode on this folder
- Create Git repository
- Create text file and write some content inside
- Commit the file on the active branch (master)
- Create and change branch from master to develop
- Modify the file
- Commit the file on the active branch (develop)
- Modify the file again
- Change branch to master **without commit**

At the end I try to click anywhere on the screen and system toolbar, but nothing is effective. I had to kill the process from another computer to get back my computer.

I reproduce with ```code --disable-extensions```

I installed vscode from the ubuntu source packages.
I tried with a fresh install from .tar.gz package downloaded on the website and have the same problem.

**Comment by joaomoreno** [comment]
Does this happen on other dialogs? Try `File > Open Folder` or `Help > About`.

**Comment by ggrimbert** [comment]
No, this is the only one blocking my screen.
But I now remember that I had a similar problem some months ago https://github.com/Microsoft/vscode/issues/50026

**Comment by joaomoreno** [comment]
Definitely sounds like an Electron issue on your specific setup.

**Comment by ggrimbert** [comment]
There is no problem with the dialog. Everything runs fine

**Comment by joaomoreno** [comment]
@bpasero Some thoughts?

**Comment by bpasero** [comment]
@ggrimbert does this happen with any dialog? e.g. try with closing a dirty editor?

**Comment by ggrimbert** [comment]
No, I haven't find any other dialog that does it except this one https://github.com/Microsoft/vscode/issues/50026 that has been fixed by removing the modal

**Comment by bpasero** [comment]
We are building exploration builds that use a much newer version of Electron (our UI framework). I wonder if this issue reproduces with one of these builds, could you try? Download:
* macOS: [Download](https://az764295.vo.msecnd.net/exploration/45498d93166483c2eff886e28f8dab915ae382fc/VSCode-darwin-exploration.zip)
* Linux: [Download](https://az764295.vo.msecnd.net/exploration/5a736c2e4904ed114fae2d5941508505f53afc66/code-exploration_1.28.0-1538127308_amd64.deb)
* Windows: [Download](https://az764295.vo.msecnd.net/exploration/5a736c2e4904ed114fae2d5941508505f53afc66/VSCodeUserSetup-x64-1.28.0-exploration.exe)

**Comment by ggrimbert** [comment]
I've tried with the .deb version downloaded from your link, and from the apt version available on my system. The same error appears.

**Comment by joaomoreno** [comment]
Can you show me a screenshot of the modal dialog from the Fiddle?

**Comment by ggrimbert** [comment]
screenshot with the modal : 
![image](https://user-images.githubusercontent.com/15869188/49570512-1f648900-f937-11e8-9425-72826ee6e6b0.png)


screenshot after clicking inside the modal
![image](https://user-images.githubusercontent.com/15869188/49570523-2a1f1e00-f937-11e8-92cc-3f69e8749020.png)

**Comment by ggrimbert** [comment]
I've tried to run vscode with the dev tools open; unfortunately I have no logs inside.

**Comment by joaomoreno** [comment]
Looks pretty similar to the VS Code one... I am clueless. :confused:

**Comment by bpasero** [comment]
@ggrimbert can you share the code you used for the fiddle? did you use async dialogs? we might be using dialogs different from what you tried in the fiddle.

**Comment by ggrimbert** [comment]
the code is inside https://gist.github.com/joaomoreno/89d54e73b90415e0024a81ed17af445d
@joaomoreno give it to me in a previous comment

**Comment by ggrimbert** [comment]
Same result as previously, everything works fine, nothing blocks. 
I tried to open the two dialogs, one after another and everything is good.

**Comment by MrPointer** [comment]
I can confirm it also happens to me on **Windows**. Same version, exact scenario.
BTW - It also happens when switching branches _outside_ of vscode, which might just hint a bug in the git extension. 
Moreover, I'm using the [CMake Tools Extension](https://github.com/vector-of-bool/vscode-cmake-tools) in my workspace and the git freeze also causes the **cmake.exe** _Windows Process_ to eat up my CPU and RAM, alongside vscode, until it all just crashes (probably because of a StackOverflow).

**Comment by EricJizbaMSFT** [comment]
I ran into this issue as well for a different dialog on Ubuntu 16.04. I tried to boil it down to the simplest case and here's my code that repros the problem every time:
```
await vscode.window.showQuickPick([{label: 'Pick me'}], {});
await vscode.window.showWarningMessage('I warn you', { modal: true}, 'Yes');
```

Here's a very simple extension with the full code (on the "modal" branch): https://github.com/EricJizbaMSFT/helloworldext/tree/modal 

The key things i noticed:
1. It only happens if I have a quick pick before the modal dialog
1. It only happens if I use my mouse to select an item from the quick pick (as opposed to hitting enter)
    1. @dhyey35 noticed the same thing in https://github.com/Microsoft/vscode/issues/66274, which is probably a dupe of this issue
